### ISSUE TITLE
ExecuteQuery for unknown # of fields

### ISSUE THREAD

**Issue by tofutim** [praeclarum/sqlite-net#82]
I have a certain scenario where I generate SQL statements on the fly - the fields inside may vary. Is there anyway I can query for this without building a class ahead of time?

I'm looking for ExecuteQuery() without the <T>.

**Comment by torfeld6** [comment]
Hi @tofutim. Thanks you for the great code

**Comment by dazinator** [comment]
Is there anyone that can get this into a new NuGet release?

**Comment by praeclarum** [comment]
Yes this does seem like a reasonable feature. I think I would prefer dynamic over dictionary though.

**Comment by MKuckert** [comment]
Should be pretty simple to leverage the [ExpandoObject](https://msdn.microsoft.com/en-us/library/system.dynamic.expandoobject(v=vs.110).aspx)

**Comment by igium** [comment]
@tofutim, I love you, man!

Putting the code in a partial SQLiteCommand implementation safely extends the class without the risk of loosing the changes after the sqlite.net module update/reinstall.

**Comment by bill2004158** [comment]
@praeclarum 

how about this? if no problem, can you add to master branch?

	public partial class SQLiteConnection
	{
		public List<ExpandoObject> Query(string query, params object[] args)
		{
			var cmd = CreateCommand (query, args);
			return cmd.ExecuteQuery();
		}
	}

	public partial class SQLiteCommand
	{
		public List<ExpandoObject> ExecuteQuery() {
			return ExecuteDeferredQuery().ToList();
		}

		private IEnumerable<ExpandoObject> ExecuteDeferredQuery () {
			if (_conn.Trace) {
				_conn.Tracer?.Invoke ("Executing Query: " + this);
			}

			var stmt = Prepare ();
			try {
				var cols = new string[SQLite3.ColumnCount (stmt)];

				for (var i = 0; i < cols.Length; i++) {
					var name = SQLite3.ColumnName16 (stmt, i);
					cols[i] = name;
				}

				while (SQLite3.Step (stmt) == SQLite3.Result.Row) {
					var obj = new ExpandoObject() as IDictionary<string, object>;
					for (var i = 0; i < cols.Length; i++) {
						var colType = SQLite3.ColumnType (stmt, i);

						var val = ReadCol (stmt, i, colType, MapToCSharpType(colType));
						obj[cols[i]] = val;
					}
					OnInstanceCreated (obj);
					yield return (ExpandoObject) obj;
				}
			}
			finally {
				SQLite3.Finalize (stmt);
			}
		}

		private Type MapToCSharpType (SQLite3.ColType colType)
		{
			Type targetType;
			switch (colType) {
				case SQLite3.ColType.Text:
					targetType = typeof (string);
					break;
				case SQLite3.ColType.Integer:
					targetType = typeof (int);
					break;
				case SQLite3.ColType.Float:
					targetType = typeof (double);
					break;
				case SQLite3.ColType.Blob:
					targetType = typeof (byte[]);
					break;
				default:
					targetType = typeof (object);
					break;
			}

			return targetType;
		}
	}

**Comment by lapoPaolacci** [comment]
with this method if I do complex query is very slow and device crash

**Comment by rogerskk** [comment]
I have a select query that I need to use a dynamic type with.  I am trying
`var result= _database.Query<Dictionary<string,object>>(query);`
This returns the correct number in the list, but the keys and values are blank.  Is there a different way I should be running this to get the dynamic results?
